File: BE/shopapp/src/main/java/com/project/shopapp/DTOs/reponses/ProductResponse.java
package com.project.shopapp.DTOs.reponses;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.shopapp.DTOs.ProductDTO;
import com.project.shopapp.models.Product;
import lombok.*;
import java.math.BigDecimal;

@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductResponse extends BaseResponse{
    private String name;
    private BigDecimal price;
    private String image;
    private String description;

    @JsonProperty("category_id")
    private Long categoryId;

//    @JsonProperty("name")
//    private String categoryName;

    public static ProductResponse fromProduct(Product product)
    {
        ProductResponse productResponse = ProductResponse.builder()
                .name(product.getName())
                .price(product.getPrice())
                .image(product.getImage())
                .description(product.getDescription())
                .categoryId(product.getCategory().getId())
                .build();
        productResponse.setCreatedAt(product.getCreatedAt());
        productResponse.setUpdatedAt(product.getUpdatedAt());
        return productResponse;
    }
}

File: FE/shopapp-angular/src/app/responses/products/product.response.ts
import { ProductImage } from "./product-image.response";

export interface ProductResponse {
    id: number;
    name: string;
    price: number;
    image: string | null; // Image có thể là string (đường dẫn) hoặc null
    description: string;
    created_at: number[]; // Mảng chứa các giá trị số nguyên đại diện cho ngày tạo
    updated_at: number[]; // Mảng chứa các giá trị số nguyên đại diện cho ngày cập nhật
    category_id: number;
    product_images: ProductImage[]; // Mảng chứa các hình ảnh của sản phẩm
    quantity: number;
    selected: boolean|false;
}